src/components/settings-modal/settings-modal.js

```diff
@@ -80,19 +90,49 @@ const SettingsModal = ({
     hasClickedApplyAndClose,
     showFeatureHintsValue,
     isPrettyNameValue,
+    showDatasetPreviewsValue,
     onToggleFlag,
     onToggleShowFeatureHints,
     onToggleIsPrettyName,
+    onToggleShowDatasetPreviews,
     showSettingsModal,
     toggleFlags,
   ]);
 
+  useEffect(() => {
+    const updatePreferences = async () => {
+      try {
+        const response = await fetch('/api/preferences', {
+          method: 'POST',
+          headers: {
+            'Content-Type': 'application/json',
+          },
+          body: JSON.stringify({
+            showDatasetPreviews: showDatasetPreviewsValue,
+          }),
+        });
+
+        if (!response.ok) {
+          throw new Error('Failed to update preferences');
+        }
+
+        const result = await response.json();
+        console.log('Preferences updated successfully:', result);
+      } catch (error) {
+        console.error('Error updating preferences:', error);
+      }
+    };
```

Comment:
Please move this api call to separate file.

Reply:
Moved it to `src/utils/preferences-api.js`.